app/src/main/java/com/ittinder/ittinder/fragment/LoginFragment.kt

```diff
@@ -1,24 +1,114 @@
 
 class LoginFragment : Fragment() {
+    private lateinit var userViewModel: UserViewModel
     private var _binding: FragmentLoginBinding? = null
     private val binding get() = _binding!!
 
+    //set LiveData listeners
+    private val emailLiveData = MutableLiveData<String>()
+    private val passwordLiveData = MutableLiveData<String>()
+    private val isValidLiveData = MediatorLiveData<Boolean>().apply {
+        this.value = false
+
+        //listen to email
+        addSource(emailLiveData) { email ->
+            val password = passwordLiveData.value
+            this.value = validateForm(email, password)
+        }
+        //listen to password
+        addSource(passwordLiveData) { password ->
+            val email = emailLiveData.value
+            this.value = validateForm(email, password)
+        }
+    }
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         _binding = FragmentLoginBinding.inflate(inflater, container, false)
+
+//        val viewModel: UserViewModel by viewModels();
+//        viewModel.user
+//        activity.findViewById(R.menu.menu)
         return binding.root
     }
 
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val pref = activity!!.getPreferences(Context.MODE_PRIVATE)
+        val userId: Long = pref.getLong("user_id", 0)
+
+        if(!userId.equals(0)){
+            findNavController().navigate(R.id.action_loginFragment_to_swipeScreen)
+        }
+
+        val emailEditText = binding.emailAddress.editText
+        val passwordEditText = binding.password.editText
+        val loginButton = binding.login
+
+        emailEditText?.doOnTextChanged { text, _, _, _ ->
+            emailLiveData.value = text?.toString()
+        }
+
+        passwordEditText?.doOnTextChanged { text, _, _, _ ->
+            passwordLiveData.value = text?.toString()
+        }
+
+        isValidLiveData.observe(this.viewLifecycleOwner) { isValid ->
+            loginButton.isEnabled = isValid
+        }
+
+        val viewModel: UserViewModel by viewModels();
+
+        loginButton.setOnClickListener {
+            viewModel.login(
+                emailEditText?.text.toString(),
+                passwordEditText?.text.toString(),
+                activity!!
+            ).observe(this.viewLifecycleOwner) { successful ->
+                if (successful) {
+                    findNavController().navigate(R.id.action_loginFragment_to_swipeScreen)
+                    Toast.makeText(context, "Success", Toast.LENGTH_SHORT).show()
+                } else {
+                    Toast.makeText(context, "Invalid credentials", Toast.LENGTH_SHORT).show()
+                }
+            }
+        }
+
+        binding.register.setOnClickListener {
+            findNavController().navigate(R.id.action_loginFragment_to_registerFragment)
```

Comment:
is zoiets als findNavController().navigate(LoginFragmentDirections.actionLoginFragmentToSwipeScreen) niet mogelijk hier op basis van de nav graph en misschien netter?